docs/user-guide/src/capm3/annotation_based_ippool.md

```diff
@@ -0,0 +1,93 @@
+# Annotation-based IPPool Reference
+
+The `fromPoolAnnotation` field enables dynamic IPPool name resolution from
+annotations on BareMetalHost, Machine, or Metal3Machine objects. This allows
+flexible IPPool assignment based on deployment-time decisions rather than
+static template configuration.
+
+## Use Cases
+
+The `fromPoolAnnotation` field can be used in both IPv4 and IPv6 network
+configurations:
+
+- Network configuration to specify which IPPool to use for IP address allocation
+- Gateway configuration to specify which IPPool to use for gateway IP resolution
+- Route gateway configuration to specify which IPPool to use for route gateway
+  IP resolution
+
+## Field Specification
+
+The `fromPoolAnnotation` field contains:
+
+- **object**: The object type to read the annotation from (`baremetalhost`,
+  `machine`, `metal3machine`)
+- **annotation**: The annotation key containing the IPPool name
+
+> **Note:** When `fromPoolAnnotation` is set, `fromIPPool` and `fromPoolRef`
```

Comment:
Shall we mention briefly also that when not set , whats the workflow so that users are aware and can choose which way to go?

Reply:
Yes thanks, I am happy with this.